screenshots/Screenshots.ts

```diff
@@ -22,21 +34,68 @@ export class Screenshots {
 
       this.directory = directory === undefined || directory === null ? '' : `${ directory }/`;
     }
+    if (opt?.isOsCommand) {
+      this.isOsCommand = opt.isOsCommand;
+    }
   }
 
-  protected setPath(title: string, opt: PageScreenshotOptions = {}) {
-    return {
-      ...opt,
-      path: `screenshots/output/${ os.platform() }/${ this.directory }${ this.screenshotIndex++ }_${ title }.png`,
-    };
+  protected buildPath(title: string): string {
+    return `screenshots/output/${ os.platform() }/${ this.directory }${ this.screenshotIndex++ }_${ title }.png`;
   }
 
-  protected async screenshot(path: string) {
-    await this.page.screenshot(this.setPath(path, { fullPage: true }));
+  protected osCommand(file: string): string {
+    if (os.platform() === 'darwin') {
+      return `screencapture -l $(GetWindowID  "${ this.appBundleTitle }" "${ this.windowTitle }") ${ file }`;
+    }
+    if (os.platform() === 'win32') {
+      return `${ path.resolve(process.cwd(), 'resources', 'ShareX', 'sharex') } -p -s -ActiveWindow`;
+    }
+
+    return `gnome-screenshot -w -f ${ file }`;
+  }
+
+  protected async screenshot(title: string) {
+    const options = {
+      fullPage: true,
+      path:     this.buildPath(title),
+    };
+
+    await this.page.screenshot(options);
+
+    if (this.isOsCommand) {
+      const command = this.osCommand(options.path);
+
+      try {
+        childProcess.execSync(command);
+
+        if (os.platform() === 'win32') {
+          // sleep to allow ShareX to write screenshot
+          await (new Promise((resolve) => {
+            setTimeout(resolve, this.sleepDuration);
+          }));
```

Comment:
My original thought was to configure a file system watcher to move and rename screenshots, but I opted for the simple solution to sleep instead.